subsys/bluetooth/mesh/adv.c

```diff
@@ -99,7 +100,15 @@ struct net_buf *bt_mesh_adv_create(enum bt_mesh_adv_type type,
 				   enum bt_mesh_adv_tag tag,
 				   uint8_t xmit, k_timeout_t timeout)
 {
-	return bt_mesh_adv_create_from_pool(&adv_buf_pool, adv_alloc, type,
+#if defined(CONFIG_BT_MESH_RELAY)
+	if (tag & BT_MESH_RELAY_ADV) {
+		return bt_mesh_adv_create_from_pool(&relay_buf_pool,
+						    adv_relay_pool, type,
+						    tag, xmit, timeout);
+	}
+#endif
+
+	return bt_mesh_adv_create_from_pool(&adv_buf_pool, adv_local_pool, type,
```

Comment:
Suggestion: Should we rename adv_buf_pool to local_buf_pool?

Reply:
I was thinking about that, but it is not used only for local messages if 1 adv set or legacy advertising API is used.